yare-model-validators/src/main/java/com/sabre/oss/yare/model/validator/DefaultRuleValidator.java

```diff
@@ -28,7 +28,13 @@
 import com.sabre.oss.yare.core.reference.ChainedTypeExtractor;
 import com.sabre.oss.yare.core.reference.PlaceholderExtractor;
 
+import java.util.*;
+
+import static java.util.Collections.singletonList;
+import static java.util.Collections.unmodifiableSet;
+
 public final class DefaultRuleValidator {
+    private static final Set<String> reservedNames = unmodifiableSet(new LinkedHashSet<>(singletonList("ctx")));
```

Comment:
What do you think about adding "ValidatorsConfig" class that would provide the DefaultRuleValidator with the Set of reserved names? Right now it's not entirely needed, but it could be after we start adding more and more reserved names

Reply:
Good point @KH8. We should move reservedNames to ReferenceValidator, together with validation logic.